dev.exs

```diff
@@ -283,19 +283,27 @@ seeds = fn ->
     """
   })
 
-  Beacon.Admin.MediaLibrary.upload(
-    "dev",
-    Path.join(:code.priv_dir(:beacon), "assets/dockyard.png"),
-    "dockyard_1.png",
-    "image/png"
-  )
-
-  Beacon.Admin.MediaLibrary.upload(
-    "dev",
-    Path.join(:code.priv_dir(:beacon), "assets/dockyard.png"),
-    "dockyard_2.jpg",
-    "image/jpg"
-  )
+  metadata =
```

Comment:
I see the benefits for testing and I agree that 5+ args is too much (it has 4 now). I'm just worried that we're exposing `FileMetadata` internals that should be private. One could write `upload(%FileMetadata{config: nil})` and break the pipeline, which I see is possible since the spec would look like `@spec upload(FileMetadata.t)`, right?

Reply:
Can you please add a `TODO: https://github.com/BeaconCMS/beacon/pull/239#discussion_r1194160478` close to `def new`? To serve as a reminder to revisit it before v0.1 so we can move forward with this PR.

Reply:
Will do!

A developer could write `upload(%FileMetadata{config: nil})` without a catchall, but it would have to go through a PR process.

If you simply prefer to have the FileMetadata constructed inside the function, that is fine and I'm happy to change it. 

Doing that though means either upload needs to accept a file size, or we need to calculate it somewhere.

Reply:
Let's keep it as it's and see how it plays as we work on Admin